pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -0,0 +1,66 @@
+package csidrivercontrollerservicecontroller
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/client-go/informers/core/v1"
+
+	opv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehash"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// WithObservedProxyHook injects into the deployment's containers the observed proxy config.
+func WithObservedProxyHook() DeploymentHookFunc {
```

Comment:
This basically assumes only deployment needs to be updated when proxy configuration changes, but what if pods managed by daemonset needs to be restarted?

Reply:
Added a hook for the DaemonSet